### PR TITLE
Fix #16 Does not work on High Sierra

### LINKED ISSUE #16: Does not work on High Sierra
It seems like `imagesnap` doesn't actually work on High Sierra. (It looks like it does, no errors, but there is no output file)

```
$ ./imagesnap -w 1 foo.jpg
Capturing image from device "<AVCaptureDALDevice: 0x7f9b7ad2f800 [FaceTime Camera][CC27173CH6UHGWCAE]>"...~
```

(I've tried with both 0.2.5 and the latest commit d947e68)

### PR DESCRIPTION
**Pull request by jchampemont**
Add a dispatch_semaphore to avoid program exitting before the file is saved to disk.

### CODE REVIEW

File: ImageSnap.m [hunk 1]
```diff
@@ -131,7 +134,7 @@ - (void)saveSingleSnapshotFrom:(AVCaptureDevice *)device
     } else {
         [self takeSnapshotWithFilename:[self fileNameWithSequenceNumber:0]];                // Capture a frame
     }
-
+    dispatch_semaphore_wait(_semaphore, DISPATCH_TIME_FOREVER);
```
Comment (gregdm) [inline ImageSnap.m]: add a timeout for the semaphore ?

**Comment by nkirsch** [comment]
This branch fixed it for me.

**Comment by LeviSchuck** [comment]
Why is this not landed yet?

**Comment by ilovezfs** [comment]
@rharder any thoughts here on merging?

**Comment by renesugar** [comment]
A separate problem is the file name passed on the command line isn't being used (see attached git diff).

[imagesnap.txt](https://github.com/rharder/imagesnap/files/2212196/imagesnap.txt)

**Pull request merged**